recipes/dns_config.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: dns_config
+#
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+# Utility function to retrieve the assigned hostname from DynamoDB
+def retrieve_assigned_hostname
+  instance_id = shell_out!('curl -s http://169.254.169.254/latest/meta-data/instance-id', user: 'root').stdout.strip
+  Chef::Log.info("Instance id is '#{instance_id}'")
+
+  assigned_hostname = shell_out!("#{node['cfncluster']['cookbook_virtualenv_path']}/bin/aws dynamodb " \
+      "--region #{node['cfncluster']['cfn_region']} query --table-name #{node['cfncluster']['stack_name']} " \
+      "--index-name InstanceId --key-condition-expression 'InstanceId = :instanceid' " \
+      "--expression-attribute-values '{\":instanceid\": {\"S\":\"#{instance_id}\"}}' " \
+      "--output text --query 'Items[0].Id.S'", user: 'root').stdout.strip
```

Comment:
just to confirm: when item is not in the table the query return None?

Reply:
Yes